pkg/cgroups/cgroupstats.go

```diff
@@ -0,0 +1,541 @@
+	}
+
+	number, err := strconv.ParseInt(lines[0], 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return number, nil
+}
+
+var blkLineRe = regexp.MustCompile(`^(\d+):(\d+) (\w+) (\d+)$`)
+
+// GetBlkioThrottleBytes returns amount of bytes transferred to/from the disk.
+func GetBlkioThrottleBytes(cgroupPath string) (BlkioThrottleBytes, error) {
+
+	// File looks like this:
+	//
+	// 8:16 Read 4223325184
+	// 8:16 Write 3207528448
+	// 8:16 Sync 5387592704
+	// 8:16 Async 2043260928
+	// 8:16 Discard 0
+	// 8:16 Total 7430853632
+	// 8:0 Read 5246572032
+	// 8:0 Write 2361737216
+	// 8:0 Sync 5575892480
+	// 8:0 Async 2032416768
+	// 8:0 Discard 0
+	// 8:0 Total 7608309248
+	// Total 15039162880
+
+	lines, err := readCgroupFileLines(path.Join(cgroupPath, "blkio.throttle.io_service_bytes_recursive"))
+
+	if err != nil {
+		return BlkioThrottleBytes{}, err
+	}
+
+	if len(lines) == 1 && lines[0] == "Total 0" {
+		return BlkioThrottleBytes{TotalBytes: 0}, nil
+	}
+
+	result := BlkioThrottleBytes{}
+	result.DeviceBytes = make([]BlkioDeviceBytes, 0)
+
+	for _, line := range lines {
+		if line[:5] == "Total" {
+			tokens := strings.Split(line, " ")
+			if len(tokens) == 2 {
+				totalBytes, err := strconv.ParseInt(tokens[1], 10, 64)
+				if err != nil {
+					return BlkioThrottleBytes{}, err
+				}
+				result.TotalBytes = totalBytes
+			}
+		} else {
+			tokens := blkLineRe.FindStringSubmatch(line)
+			if tokens == nil || len(tokens) != 5 {
+				return BlkioThrottleBytes{}, fmt.Errorf("error parsing file")
+			}
+			major64, err := strconv.ParseInt(string(tokens[1]), 10, 32)
+			if err != nil {
+				return BlkioThrottleBytes{}, err
+			}
+			minor64, err := strconv.ParseInt(string(tokens[2]), 10, 32)
+			if err != nil {
+				return BlkioThrottleBytes{}, err
+			}
+			major := int(major64)
+			minor := int(minor64)
+
+			var device *BlkioDeviceBytes
+			// Find if we already have the device. There should only be a few so this isn't too costly.
+			for _, dev := range result.DeviceBytes {
+				if dev.Major == major && dev.Minor == minor {
+					device = &dev
+				}
+			}
+
+			if device == nil {
+				device = &BlkioDeviceBytes{Major: major, Minor: minor, Operations: make(map[string]int64)}
+				result.DeviceBytes = append(result.DeviceBytes, *device)
+			}
+			bytes, err := strconv.ParseInt(string(tokens[4]), 10, 64)
+			if err != nil {
+				return BlkioThrottleBytes{}, err
+			}
+
+			device.Operations[string(tokens[3])] = bytes
+		}
+	}
+
+	return result, nil
+}
+
+// GetCPUAcctStats retrieves CPU account statistics for a given cgroup.
+func GetCPUAcctStats(cgroupPath string) ([]CPUAcctUsage, error) {
+
+	// File looks like this:
+	//
+	// cpu user system
+	// 0 3723082232186 2456599218
+	// 1 3748398003001 1149546796
+
+	lines, err := readCgroupFileLines(path.Join(cgroupPath, "cpuacct.usage_all"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]CPUAcctUsage, len(lines)-1)
+
+	for i, line := range lines {
```

Comment:
just a suggestion:
```
for i, line := range lines[1:]
```
would make the logic in the loop slightly simpler

Reply:
Updated. Also further simplified the loop by doing filtering with continue's instead of nested if's. 
Also fixed the combo bug of incorrectly sizing result + the loop index i not reflecting skipped lines which could cause zero-valued 'holes' or trailing items in result if lines were really skipped.